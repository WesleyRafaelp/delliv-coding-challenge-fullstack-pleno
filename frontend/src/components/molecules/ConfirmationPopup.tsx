import React from 'react';

interface ConfirmationPopupProps {
  isOpen: boolean;
  onCancel: () => void;
  onConfirm: () => void;
}

const ConfirmationPopup: React.FC<ConfirmationPopupProps> = ({ isOpen, onCancel, onConfirm }) => {
  if (!isOpen) {
    return null;
  }

  return (
    <div className="fixed top-0 left-0 w-full h-full bg-gray-800 bg-opacity-50 flex justify-center items-center">
      <div className="bg-white p-4 rounded-md">
        <p>VocÃª tem certeza que deseja excluir este pedido?</p>
        <div className="flex justify-end mt-4">
          <button onClick={onCancel} className="mr-2">
            Cancelar
          </button>
          <button onClick={onConfirm} className="text-red-500 hover:text-red-700">
            Confirmar
          </button>
        </div>
      </div>
    </div>
  );
};

export default ConfirmationPopup;
