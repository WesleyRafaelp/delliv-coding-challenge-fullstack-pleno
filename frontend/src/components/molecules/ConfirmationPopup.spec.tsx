import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import ConfirmationPopup from './ConfirmationPopup';

describe('ConfirmationPopup Component', () => {
  it('renders ConfirmationPopup when isOpen is true', () => {
    render(
      <ConfirmationPopup isOpen={true} onCancel={jest.fn()} onConfirm={jest.fn()} />
    );

    const popupElement = screen.getByText(/Você tem certeza que deseja excluir este pedido?/i);
    expect(popupElement).toBeInTheDocument();
  });

  it('does not render ConfirmationPopup when isOpen is false', () => {
    render(
      <ConfirmationPopup isOpen={false} onCancel={jest.fn()} onConfirm={jest.fn()} />
    );

    const popupElement = screen.queryByText(/Você tem certeza que deseja excluir este pedido?/i);
    expect(popupElement).not.toBeInTheDocument();
  });

  it('calls onCancel when Cancelar button is clicked', () => {
    const onCancelMock = jest.fn();
    render(<ConfirmationPopup isOpen={true} onCancel={onCancelMock} onConfirm={jest.fn()} />);

    const cancelButton = screen.getByText('Cancelar');
    fireEvent.click(cancelButton);

    expect(onCancelMock).toHaveBeenCalled();
  });

  it('calls onConfirm when Confirmar button is clicked', () => {
    const onConfirmMock = jest.fn();
    render(<ConfirmationPopup isOpen={true} onCancel={jest.fn()} onConfirm={onConfirmMock} />);

    const confirmButton = screen.getByText('Confirmar');
    fireEvent.click(confirmButton);

    expect(onConfirmMock).toHaveBeenCalled();
  });

  it('matches snapshot when isOpen is true', () => {
    const { asFragment } = render(
      <ConfirmationPopup isOpen={true} onCancel={jest.fn()} onConfirm={jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot when isOpen is false', () => {
    const { asFragment } = render(
      <ConfirmationPopup isOpen={false} onCancel={jest.fn()} onConfirm={jest.fn()} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

});
