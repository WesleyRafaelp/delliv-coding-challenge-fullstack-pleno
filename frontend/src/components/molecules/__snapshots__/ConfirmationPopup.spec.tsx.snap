// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationPopup Component matches snapshot when isOpen is false 1`] = `<DocumentFragment />`;

exports[`ConfirmationPopup Component matches snapshot when isOpen is true 1`] = `
<DocumentFragment>
  <div
    class="fixed top-0 left-0 w-full h-full bg-gray-800 bg-opacity-50 flex justify-center items-center"
  >
    <div
      class="bg-white p-4 rounded-md"
    >
      <p>
        VocÃª tem certeza que deseja excluir este pedido?
      </p>
      <div
        class="flex justify-end mt-4"
      >
        <button
          class="mr-2"
        >
          Cancelar
        </button>
        <button
          class="text-red-500 hover:text-red-700"
        >
          Confirmar
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
